internal/configsource/yamlconfigsource/README.md

```diff
@@ -0,0 +1,53 @@
+# YAML Config Source (Alpha)
+
+Use the YAML config source to inject [golang templates](https://pkg.go.dev/text/template)
+into the configuration.
+
+## Configuration
+
+Under the `config_sources:` use `yaml:` or `yaml/<name>:` to create a
+YAML config source.
+
+```yaml
+config_sources:
+  yaml:
+```
+
+The parameters of an YAML config source are passed to template to be processed.
+For example, assumint that `./templates/component_template` looks like:
+
+```terminal
+logs_path: {{ .glob_pattern }}
+log_format: {{ .format }}
+```
+
+Given the configuration file:
+
+```yaml
+config_sources:
+  yaml:
+
+components:
+  # component_0 is built from the ./templates/component_template file
+  # according to the template parameters and commands. The example below
+  # defines a few parameters to be used by the template.
+  component_0: |
+    $yaml: ./templates/component_template
+    glob_pattern: /var/**/*.log
+    format: json
```

Comment:
Not directly related to this PR but could we do multi-param like:

```yaml
    $yaml:
       selector: ./templates/component_template
       glob_pattern: /var/**/*.log
       format: json
```

(Downside of having it within text block is the yaml inside text doesn't get syntax highlighted, etc. when editing)

Reply:
We had some early interactions with it as actual YAML. That ended up being dropped in favor of consistency when used in the single-line format. Actual YAML on single-line didn't look very good and couldn't be concatenated to other values. That said we could have the multiline format being YAML while the single line was our own but that requires a lot more code.

/cc @tigrannajaryan

Reply:
> @tigrannajaryan
> 
> > Would include be a better name?
> 
> I like `template` because all golang template processing is available but no strong opinion in this regard. Let me know if you prefer `include`.

I think I prefer `include` since it may be easier to understand for end users the purpose of it. Not strongly opposed to `template` but it appears a bit obscure, not self-explanatory what it does.

> 
> > re: stanza plugins
> 
> Interesting stuff: at first glance, it seems that they parse the YAML and extract a template from the comments. It should be doable to convert those to the templates supported here.

Yes, I think we can delete the top portion of their YAMLs, which just declare the parameters (I think we can just replace it by comments, I don't think it enables any functionality) and slightly adjust the `pipeline` portion to match the `operators` expected by `filelog` and it should work.

Reply:
(Sorry, replied to a wrong comment thread).

Reply:
np, I will rename this config source to `include`.